### PR TITLE
dev, v2.1, v3.0: add syntax descriptions about sql statements

### PR DESCRIPTION
**Pull request by yikeke**
### What is changed, added or deleted? 

This PR is to align v3.0 changes from https://github.com/pingcap/docs-cn/pull/1451/ and https://github.com/pingcap/docs-cn/pull/1416/. I added a new section about the syntax descriptions in create-table.md and create-database.md. @zimulala @lilin90 PTAL 

### What is the related PR or file link(s)? 

Ref: 
https://github.com/pingcap/docs-cn/pull/1451/
https://github.com/pingcap/docs-cn/pull/1416/

### Which version does your change affect? 

dev, v3.0, v2.1

### Checklist 

- [ ] Add a new file to `TOC.md`
- [ ] No Tab spaces anywhere 
- [ ] Leave a blank line both before and after a code block
- [ ] Keep the first level heading consistent with `title` in metadata
- [ ] Use *four* spaces for each level of indentation except that in `TOC.md`

### CODE REVIEW

File: dev/reference/sql/statements/create-database.md [hunk 1]
```diff
@@ -30,6 +30,23 @@ category: reference
...
+    [create_specification] ...
+
+create_specification:
+    [DEFAULT] CHARACTER SET [=] charset_name
+  | [DEFAULT] COLLATE [=] collation_name
+```
+
+当创建已存在的数据库且不指定使用 `IF NOT EXISTS` 时会报错。
+
+`create_specification` 选项用于指定数据库具体的 `CHARACTER SET` 和 `COLLATE`。目前 TiDB 只支持部分的字符集和排序规则，请参照[字符集支持](/dev/reference/sql/character-set.md)。
```
Comment (lilin90) [inline dev/reference/sql/statements/create-database.md]:
```suggestion
`create_specification` 选项用于指定数据库具体的 `CHARACTER SET` 和 `COLLATE`。目前 TiDB 只支持部分的字符集和排序规则，请参照[字符集支持](/reference/sql/character-set.md)。
```

File: dev/reference/sql/statements/create-table.md [hunk 2]
```diff
@@ -54,6 +54,171 @@ category: reference
...
+  | DELAY_KEY_WRITE [=] {0 | 1}
+  | ENGINE [=] engine_name
+  | KEY_BLOCK_SIZE [=] value
+  | MAX_ROWS [=] value
+  | MIN_ROWS [=] value
+  | ROW_FORMAT [=] {DEFAULT|DYNAMIC|FIXED|COMPRESSED|REDUNDANT|COMPACT}
+  | STATS_PERSISTENT [=] {DEFAULT|0|1}
+```
+
+`table_option` 目前支持的只有 `AUTO_INCREMENT`、`SHARD_ROW_ID_BITS`（详情介绍请参考 [TiDB 专用系统变量和语法](/dev/reference/configuration/tidb-server/tidb-specific-variables.md#shard_row_id_bits)）、`PRE_SPLIT_REGIONS`、`CHARACTER SET`、`COLLATE`  和 `COMMENT`，其它只是语法上支持。具体内容参考下表，各个子句之间用逗号隔开。
```
Comment (lilin90) [inline dev/reference/sql/statements/create-table.md]:
```suggestion
`table_option` 目前支持的只有 `AUTO_INCREMENT`、`SHARD_ROW_ID_BITS`（详情介绍请参考 [TiDB 专用系统变量和语法](/reference/configuration/tidb-server/tidb-specific-variables.md#shard-row-id-bits)）、`PRE_SPLIT_REGIONS`、`CHARACTER SET`、`COLLATE` 和 `COMMENT`，其它只是语法上支持。具体内容参考下表，各个子句之间用逗号隔开。
```

File: v2.1/reference/sql/statements/drop-database.md [hunk 3]
```diff
@@ -26,6 +26,10 @@ category: reference
 
 ![DBName](/media/sqlgram/DBName.png)
 
+## 语法说明
+
```
Comment (lilin90) [inline v2.1/reference/sql/statements/drop-database.md]: The content here is missing.

File: v3.0/reference/sql/statements/create-database.md [hunk 4]
```diff
@@ -30,6 +30,23 @@ category: reference
...
+    [create_specification] ...
+
+create_specification:
+    [DEFAULT] CHARACTER SET [=] charset_name
+  | [DEFAULT] COLLATE [=] collation_name
+```
+
+当创建已存在的数据库且不指定使用 `IF NOT EXISTS` 时会报错。
+
+`create_specification` 选项用于指定数据库具体的 `CHARACTER SET` 和 `COLLATE`。目前 TiDB 只支持部分的字符集和排序规则，请参照[字符集支持](/dev/reference/sql/character-set.md)。
```
Comment (lilin90) [inline v3.0/reference/sql/statements/create-database.md]:
```suggestion
`create_specification` 选项用于指定数据库具体的 `CHARACTER SET` 和 `COLLATE`。目前 TiDB 只支持部分的字符集和排序规则，请参照[字符集支持](/reference/sql/character-set.md)。
```

File: v3.0/reference/sql/statements/create-table.md [hunk 5]
```diff
@@ -55,6 +55,171 @@ aliases: ['/docs-cn/sql/ddl/','/docs-cn/dev/reference/sql/statements/ddl/']
...
+  | DELAY_KEY_WRITE [=] {0 | 1}
+  | ENGINE [=] engine_name
+  | KEY_BLOCK_SIZE [=] value
+  | MAX_ROWS [=] value
+  | MIN_ROWS [=] value
+  | ROW_FORMAT [=] {DEFAULT|DYNAMIC|FIXED|COMPRESSED|REDUNDANT|COMPACT}
+  | STATS_PERSISTENT [=] {DEFAULT|0|1}
+```
+
+`table_option` 目前支持的只有 `AUTO_INCREMENT`、`SHARD_ROW_ID_BITS`（详情介绍请参考 [TiDB 专用系统变量和语法](/dev/reference/configuration/tidb-server/tidb-specific-variables.md#shard_row_id_bits)）、`PRE_SPLIT_REGIONS`、`CHARACTER SET`、`COLLATE`  和 `COMMENT`，其它只是语法上支持。具体内容参考下表，各个子句之间用逗号隔开。
```
Comment (lilin90) [inline v3.0/reference/sql/statements/create-table.md]:
```suggestion
`table_option` 目前支持的只有 `AUTO_INCREMENT`、`SHARD_ROW_ID_BITS`（详情介绍请参考 [TiDB 专用系统变量和语法](/reference/configuration/tidb-server/tidb-specific-variables.md#shard-row-id-bits)）、`PRE_SPLIT_REGIONS`、`CHARACTER SET`、`COLLATE` 和 `COMMENT`，其它只是语法上支持。具体内容参考下表，各个子句之间用逗号隔开。
```

**Comment by yikeke** [comment]
Updated. PTAL again. @lilin90

File: v2.1/reference/sql/statements/create-table.md [hunk 6]
```diff
@@ -54,6 +54,164 @@ category: reference
...
+  | DELAY_KEY_WRITE [=] {0 | 1}
+  | ENGINE [=] engine_name
+  | KEY_BLOCK_SIZE [=] value
+  | MAX_ROWS [=] value
+  | MIN_ROWS [=] value
+  | ROW_FORMAT [=] {DEFAULT|DYNAMIC|FIXED|COMPRESSED|REDUNDANT|COMPACT}
+  | STATS_PERSISTENT [=] {DEFAULT|0|1}
+```
+
+`table_option` 目前支持的只有 `AUTO_INCREMENT`、`CHARACTER SET` 和 `COMMENT`，其它只是语法上支持。具体内容参考下表，各个子句之间用逗号隔开。
```
Comment (zimulala) [inline v2.1/reference/sql/statements/create-table.md]: 可以添加一下 `SHARD_ROW_ID_BITS` 和 `COLLATE` 的描述。还有 `PRE_SPLIT_REGIONS` 最新的 2.1 也支持了，是否需要加一下？

Comment (yikeke) [reply v2.1/reference/sql/statements/create-table.md]: SHARD_ROW_ID_BITS、COLLATE、PRE_SPLIT_REGIONS 在 2.1 里都支持了是吗？@zimulala

Comment (yikeke) [reply v2.1/reference/sql/statements/create-table.md]: Updated. PTAL again.

Comment (zimulala) [reply v2.1/reference/sql/statements/create-table.md]: 嗯，在最新的 v2.1 支持了。

Comment (yikeke) [reply v2.1/reference/sql/statements/create-table.md]: 最新的 v2.1 是指 2.1.13 版本之后都支持吗？@zimulala 这种情况我们希望在相应信息后面加上一个具体说明是哪个 2.1 版本的版本标签

Comment (zimulala) [reply v2.1/reference/sql/statements/create-table.md]: `COLLATE` 挺早之前就有了，`SHARD_ROW_ID_BITS` 也挺早有的，但是这个功能和 `PRE_SPLIT_REGIONS` 对于v2.1 来说都是在 2.1.13 有的。

File: v2.1/reference/sql/statements/create-table.md [hunk 7]
```diff
@@ -54,6 +54,167 @@ category: reference
...
+  | DELAY_KEY_WRITE [=] {0 | 1}
+  | ENGINE [=] engine_name
+  | KEY_BLOCK_SIZE [=] value
+  | MAX_ROWS [=] value
+  | MIN_ROWS [=] value
+  | ROW_FORMAT [=] {DEFAULT|DYNAMIC|FIXED|COMPRESSED|REDUNDANT|COMPACT}
+  | STATS_PERSISTENT [=] {DEFAULT|0|1}
+```
+
+`table_option` 目前支持的只有 `AUTO_INCREMENT`、`SHARD_ROW_ID_BITS`（详情介绍请参考 [TiDB 专用系统变量和语法](/reference/configuration/tidb-server/tidb-specific-variables.md#shard-row-id-bits)）、`PRE_SPLIT_REGIONS`、`CHARACTER SET`、`COLLATE` 和 `COMMENT`，其它只是语法上支持。具体内容参考下表，各个子句之间用逗号隔开。
```
Comment (yikeke) [inline v2.1/reference/sql/statements/create-table.md]:
```suggestion
`table_option` 目前支持的只有 `AUTO_INCREMENT`、`SHARD_ROW_ID_BITS`（详情介绍请参考 [TiDB 专用系统变量和语法](/reference/configuration/tidb-server/tidb-specific-variables.md#shard-row-id-bits)）、`PRE_SPLIT_REGIONS`、`CHARACTER SET`、`COLLATE` 和 `COMMENT`，其它只是语法上支持。具体内容参考下表，各个子句之间用逗号隔开。其中，`SHARD_ROW_ID_BITS`、`PRE_SPLIT_REGIONS 和 `COLLATE` 这三个功能都是在 TiDB 2.1.13 后开始支持。
```

File: dev/reference/sql/statements/create-table.md [hunk 8]
```diff
@@ -54,6 +54,171 @@ category: reference
...
+|----------------|--------------------------------------|----------------------------|
+|`AUTO_INCREMENT`|自增字段初始值                        |`AUTO_INCREMENT` = 5|
+|`SHARD_ROW_ID_BITS`|用来设置隐式 _tidb_rowid 的分片数量的 bit 位数 |`SHARD_ROW_ID_BITS` = 4|
+|`PRE_SPLIT_REGIONS`|用来在建表时预先均匀切分 2^(PRE_SPLIT_REGIONS-1) 个 Region |`PRE_SPLIT_REGIONS` = 4|
+|`CHARACTER SET` |指定该表所使用的字符集                | `CHARACTER SET` = 'utf8mb4'|
+|`COLLATE`       |指定该表所使用的字符集排序规则        | `COLLATE` = 'utf8mb4_bin'|
+|`COMMENT`       |注释信息                              | `COMMENT` = 'comment info'|
+
+`split-table` 配置项默认情况下会开启，在此配置项开启时，建表操作会为每个表建立单独的 Region。
+
```
Comment (yikeke) [inline dev/reference/sql/statements/create-table.md]:
```suggestion

> **注意：**
>
> 在 TiDB 2.1 版本中，`SHARD_ROW_ID_BITS`、`PRE_SPLIT_REGIONS 和 `COLLATE` 这三个功能都是从 2.1.13 版本（包括 2.1.13）开始支持的。

```

File: dev/reference/sql/statements/create-table.md [hunk 9]
```diff
@@ -54,6 +54,175 @@ category: reference
...
+> **注意：**
+>
+> 在 TiDB 2.1 版本中，`SHARD_ROW_ID_BITS`、`PRE_SPLIT_REGIONS 和 `COLLATE` 这三个功能都是从 2.1.13 版本（包括 2.1.13）开始支持的。
```
Comment (yikeke) [inline dev/reference/sql/statements/create-table.md]:
```suggestion
> 在 TiDB 2.1 版本中，`SHARD_ROW_ID_BITS`、`PRE_SPLIT_REGIONS` 和 `COLLATE` 这三个功能都是从 2.1.13 版本（包括 2.1.13）开始支持的。
```

File: dev/reference/sql/statements/create-table.md [hunk 10]
```diff
@@ -54,6 +54,175 @@ category: reference
...
+|`AUTO_INCREMENT`|自增字段初始值                        |`AUTO_INCREMENT` = 5|
+|`SHARD_ROW_ID_BITS`|用来设置隐式 _tidb_rowid 的分片数量的 bit 位数 |`SHARD_ROW_ID_BITS` = 4|
+|`PRE_SPLIT_REGIONS`|用来在建表时预先均匀切分 2^(PRE_SPLIT_REGIONS-1) 个 Region |`PRE_SPLIT_REGIONS` = 4|
+|`CHARACTER SET` |指定该表所使用的字符集                | `CHARACTER SET` = 'utf8mb4'|
+|`COLLATE`       |指定该表所使用的字符集排序规则        | `COLLATE` = 'utf8mb4_bin'|
+|`COMMENT`       |注释信息                              | `COMMENT` = 'comment info'|
+
+`split-table` 配置项默认情况下会开启，在此配置项开启时，建表操作会为每个表建立单独的 Region。
+
+> **注意：**
```
Comment (lilin90) [inline dev/reference/sql/statements/create-table.md]: Please also add this note to `v3.0` and `v2.1`.

Comment (yikeke) [reply dev/reference/sql/statements/create-table.md]: This note is only related to v2.1. I think we should move this note to v2.1 folder.

Comment (lilin90) [reply dev/reference/sql/statements/create-table.md]: @yikeke Agree with you. Please move it.

Comment (yikeke) [reply dev/reference/sql/statements/create-table.md]: I‘ve moved it, PTAL again~

File: dev/reference/sql/statements/create-database.md [hunk 11]
```diff
@@ -59,3 +76,4 @@ mysql> SHOW TABLES;
 * [USE](/reference/sql/statements/use.md)
 * [ALTER DATABASE](/reference/sql/statements/alter-database.md)
 * [DROP DATABASE](/reference/sql/statements/drop-database.md)
+* [SHOW CREATE DATABASE](/reference/sql/statements/show-create-database.md)
```
Comment (lilin90) [inline dev/reference/sql/statements/create-database.md]: This is a broken link at the website. Please help check and fix it. Thanks!

**Pull request merged**